modules/client/src/main/java/org/apache/ignite/jdbc/JdbcStatement.java

```diff
@@ -0,0 +1,647 @@
+
+        if (res.status() != JdbcResponse.STATUS_SUCCESS)
+            throw IgniteQueryErrorCode.createJdbcSqlException(res.err(), res.status());
+
+        for (JdbcQuerySingleResult jdbcRes : res.results()) {
+            if (jdbcRes.status() != JdbcResponse.STATUS_SUCCESS)
+                throw IgniteQueryErrorCode.createJdbcSqlException(jdbcRes.err(), jdbcRes.status());
+        }
+
+        resSets = new ArrayList<>(res.results().size());
+
+        for (JdbcQuerySingleResult jdbcRes : res.results()) {
+            resSets.add(new JdbcResultSet(this, jdbcRes.cursorId(), pageSize,
+                jdbcRes.last(), jdbcRes.items(), jdbcRes.isQuery(), false, jdbcRes.updateCount(),
+                closeOnCompletion, conn.handler()));
+        }
+
+        assert !resSets.isEmpty() : "At least one results set is expected";
+    }
+
+    /** {@inheritDoc} */
+    @Override public int executeUpdate(String sql) throws SQLException {
+        execute0(sql, null);
+
+        int res = getUpdateCount();
+
+        if (res == -1) {
+            closeResults();
+            throw new SQLException("The query is not DML statement: " + sql);
+        }
+
+        return res;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void close() throws SQLException {
+        if (isClosed())
+            return;
+
+        try {
+            closeResults();
+
+            conn.removeStatement(this);
+        }
+        finally {
+            closed = true;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public int getMaxFieldSize() throws SQLException {
+        ensureNotClosed();
+
+        return 0;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setMaxFieldSize(int max) throws SQLException {
+        ensureNotClosed();
+
+        if (max < 0)
+            throw new SQLException("Invalid field limit.");
+
+        throw new SQLFeatureNotSupportedException("Field size limitation is not supported.");
+    }
+
+    /** {@inheritDoc} */
+    @Override public int getMaxRows() throws SQLException {
+        ensureNotClosed();
+
+        return maxRows;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setMaxRows(int maxRows) throws SQLException {
+        ensureNotClosed();
+
+        if (maxRows < 0)
+            throw new SQLException("Invalid max rows value.");
+
+        this.maxRows = maxRows;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setEscapeProcessing(boolean enable) throws SQLException {
+        ensureNotClosed();
+    }
+
+    /** {@inheritDoc} */
+    @Override public int getQueryTimeout() throws SQLException {
+        ensureNotClosed();
+
+        return timeout / 1000;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setQueryTimeout(int timeout) throws SQLException {
+        ensureNotClosed();
+
+        if (timeout < 0)
+            throw new SQLException("Invalid timeout value.");
+
+        timeout(timeout * 1000 > timeout ? timeout * 1000 : Integer.MAX_VALUE);
```

Comment:
Zero timeout will be converted to MAX_VALUE.
Is it expected? if so, the it worth to be commented.

Reply:
Yes, Integer.MAX_VALUE means no restrictions. I added comment about this.